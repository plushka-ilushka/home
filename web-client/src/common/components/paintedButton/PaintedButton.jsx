import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './paintedButton.scss';

export default function PaintedButton(props) {

    const {
        className,
        type,
        onClick,
        caption,
        ...setProps
    } = props;

    return (
        <button
            className={ classNames("painted-button", className) }
            type={type}
            onClick={onClick}
            {...setProps}
        >
            {caption}
        </button>
    );
};

PaintedButton.propTypes = {
    className: PropTypes.string,
    type: PropTypes.string,
    onClick: PropTypes.bool,
    caption: PropTypes.string
};

PaintedButton.defaultProps = {
    type: "button",
    className: undefined,
    caption: undefined
};
