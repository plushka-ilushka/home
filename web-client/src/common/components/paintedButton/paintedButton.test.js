import React from 'react';
import { mount } from 'enzyme';
import PaintedButton from './PaintedButton';

describe('PaintedButton Component', () => {
    it('PaintedButton Snapshot', () => {
        expect(mount(<PaintedButton/>)).toMatchSnapshot();
    })
})