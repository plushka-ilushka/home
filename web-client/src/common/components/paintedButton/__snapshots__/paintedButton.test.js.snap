// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaintedButton Component PaintedButton Snapshot 1`] = `
<PaintedButton
  type="button"
>
  <button
    className="painted-button"
    type="button"
  />
</PaintedButton>
`;
